apps/emqx/integration_test/emqx_persistent_session_ds_SUITE.erl

```diff
@@ -71,13 +71,37 @@ init_per_testcase(t_session_gc = TestCase, Config) ->
         {gc_interval, timer:seconds(2)}
         | Config
     ];
+init_per_testcase(t_message_gc = TestCase, Config) ->
```

Comment:
I think there's value in having an integration test like this, see [this other comment](https://github.com/emqx/emqx/pull/12338#discussion_r1456000119) about it uncovering problems related to an ongoing iteration over a subscription.

I agree that **also** having some simpler unit tests is valuable, for sure.

Reply:
Also, if we're talking about the particular property of "replicants don't run GC", I think it might be tricky to check with a unit test if a replicant node is not spawned.  :thinking:

Reply:
My point is that this logic is implemented simply by calling `mria_membership:running_core_nodelist()` function instead of `erlang:nodes()`... I can easily believe that this property holds without any test whatsoever, let alone without spawning a cluster of real nodes :wink: 

I would rather see more tests involving `add|delete|list_generation` APIs that verify their return values and the side effects.

Reply:
Ok.

About the API tests, there's `emqx_ds_SUITE:t_08_smoke_list_drop_generation` which hopefully covers their effects.

Reply:
I'm converting the integration test to tests in this latter suite.